pontoon/batch/views.py

```diff
@@ -31,6 +31,9 @@ def mark_changed_translation(changed_entities, locale):
     changed_entities_array = []
     existing = ChangedEntityLocale.objects.values_list("entity", "locale").distinct()
     for changed_entity in changed_entities:
+        if changed_entity.resource.project.data_source == Project.DataSource.DATABASE:
```

Comment:
Same as below, let's make sure we don't query the DB for each translation by prefetching project data here:
https://github.com/mozilla/pontoon/blob/f3727dd6afe8d621d61d2a4311818d5781dc517d/pontoon/batch/utils.py#L37